demo/Demo.hpp

```diff
@@ -0,0 +1,130 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+#ifndef PEGASUS_DEMO_HPP
+#define PEGASUS_DEMO_HPP
+
+#include <pegasus/ParticleWorld.hpp>
+#include <demo/Render.hpp>
+
+namespace pegasus
+{
+/**
+ * @brief Represents a singletone instance of the Demo
+ */
+class Demo
+{
+public:
+    struct Object;
+
+    /**
+     * @brief Returns a reference to the singletone Demo instance
+     * @return Demo object reference
+     */
+    static Demo& GetInstance();
+
+    /**
+     * @brief Checks if the window is not closed and render is initialized
+     * @return @c true if initialize, @c false otherwise
+     */
+    bool IsValid() const;
+
+    /**
+     * @brief Computes and renders frame
+     */
+    void RunFrame();
+
+    /**
+     * @brief Makes physical and render instance of the line and returns it
+     * @param[in] particle physical data
+     * @param[in] start first point of the line
```

Comment:
`first point of the line` -> `line start`

Reply:
Any particular reason?

Reply:
Just wording